File: tools/bench/answers/python_sonnet/s3/01_rle.py
s = "aaabbcccca"
result = ""
i = 0
while i < len(s):
    ch = s[i]
    count = 0
    while i < len(s) and s[i] == ch:
        count += 1
        i += 1
    result += ch + str(count)
print(result)

File: tools/bench/answers/rust_haiku/s2/01_rle.rs
fn main() {
    let s = "aaabbcccca";
    let mut result = String::new();
    let mut chars = s.chars().peekable();

    while let Some(ch) = chars.next() {
        let mut count = 1;
        while let Some(&next_ch) = chars.peek() {
            if next_ch == ch {
                count += 1;
                chars.next();
            } else {
                break;
            }
        }
        result.push(ch);
        result.push_str(&count.to_string());
    }

    println!("{}", result);
}
